src/lib/schedule.c

```diff
@@ -51,6 +51,7 @@ struct schedule_data {
 	struct list_item list;	/* list of tasks in priority queue */
 	uint32_t clock;
 	struct work work;
+	int irq;
```

Comment:
net loss of 4 bytes for no gain ?

Reply:
This is needed, because interrupts will become virtual later. They will be mapped at run-time based on their interrupt controller parent and the physical interrupt number on that controller. So, we want to map interrupts once and then use the virtual IRQ number. See patch https://github.com/thesofproject/sof/pull/1053/commits/493b086682d398b95fb76ab5095f8049c57a0d43

Reply:
Ok, but is the virtual number constant for each platform. i.e. APL virt IRQ 10 is always SSP0 ? If not then this will be painful to debug.

Reply:
As long as the board configuration doesn't change, yes, they will stay constant. The only thing, that can change them is adding new or removing existing interrupt controllers. I don't think that would happen too often. We can add trace to the mapping function, as long as the IPC interrupt is a DSP native one. Or we can dump a mapping at the end of booting and then print any new mappings.